app/lib/actions/katello/provider/update.rb

```diff
@@ -15,20 +15,21 @@ def humanized_name
         private
 
         def update_url(provider, base_url)
+          current_base_url = provider.repository_url
           provider.update_attributes!(:repository_url => base_url)
 
           if provider.redhat_provider?
             provider.products.enabled.each do |product|
-              update_repository_urls(product, base_url)
+              update_repository_urls(product, current_base_url, base_url)
             end
           end
         end
 
-        def update_repository_urls(product, base_url)
+        def update_repository_urls(product, current_base_url, new_base_url)
           product.repositories.each do |repository|
             next unless repository.url
-            uri = URI.parse(repository.url)
-            url = "#{base_url}#{uri.path}"
+            path = repository.url.split(current_base_url)[1]
```

Comment:
What do you mean? So the problem lies in the fact that you could update the CDN url to be something like:

```
https://fake.com/pub/sat-import
```

Which mean sunless we strip off the entirety of the "previous" CDN URL, you can end up with the repository URL being:

```
https://cdn.redhat.com/pub/sat-import/dist/content.....
```